### PR TITLE
More beta fixes

### PR DESCRIPTION
**Pull request by thellmund**
This pull request fixes some problems encountered in v2.0 beta 3 and 4: 
1. Introduces `EduroamFixCardViewHolder` and `NextLectureCardViewHolder` for greater stability and easier-to-understand code.
2. Updates the design of `NextLectureCard`, as the buttons on the right-hand side were both hard to hit (very small) and interfered with the options button. Now, we display a “show x more ...” button so that the user can expand the card and see up to four upcoming lectures.

---

Collapsed: 
screenshot_20181005-212644

Expanded: 
screenshot_20181005-212650

### CODE REVIEW

**Comment by thellmund** [comment]
@kordianbruck @pfent @jacqueline8711 I’m not sure about the design yet. Maybe divider lines would help. What’s your take?

**Comment by jacqueline8711** [comment]
@thellmund Yes, the expanded card looks kind of weird to me with the 'Hide 1 more...' in the middle.
Maybe somewhat like this?

screen shot 2018-10-07 at 09 27 22
screen shot 2018-10-07 at 09 27 32

or with the expand / collapse action at the bottom

screen shot 2018-10-07 at 09 27 43
screen shot 2018-10-07 at 09 27 51

**Comment by thellmund** [comment]
@jacqueline8711 I like the design ideas. Let me know what you think of the current state that I’ve just pushed.

**Comment by thellmund** [comment]
The accidental renaming of "card" to "cardView" in many comments is also reverted.

**Comment by jacqueline8711** [comment]
I think we should keep the link to the Roomfinder in the NextLectureCard, it's quite useful if you need to find your room quickly

**Comment by thellmund** [comment]
I addressed the bug in `NextLectureCard`. 

Regarding the room finder: I’m now directly opening the event fragment within the `CalendarActivity` when tapping on the event in the card, removing one unnecessary step from the flow. I don’t think a direct link to the room finder is necessary: I strongly assume that the assumption that users need the room finder frequently is wrong.

**Comment by jacqueline8711** [comment]
Yes, the roomfinder is probably mostly needed for the first week of the semester. At least I need it a lot then.
Opening the fragment directly is definitely better. It's only working for the first event though, can we make it work for the other ones as well?

File: .idea/codeStyleSettings.xml [hunk 1]
```diff
@@ -1,264 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jacqueline8711) [inline .idea/codeStyleSettings.xml]: We probably want to keep the .idea files?

**Comment by thellmund** [comment]
> Yes, the roomfinder is probably mostly needed for the first week of the semester. At least I need it a lot then.
> Opening the fragment directly is definitely better. It's only working for the first event though, can we make it work for the other ones as well?

Fixed.

File: .idea/codeStyleSettings.xml (hunk 1 above)
Comment (thellmund) [reply .idea/codeStyleSettings.xml]: Yes, Android Studio keeps doing that automatically somehow. Will revert it.

File: app/src/main/res/layout/layout_lecture_selection.xml [hunk 2]
```diff
@@ -6,58 +6,55 @@
...
-            android:layout_width="@dimen/next_lecture_select_width"
-            android:layout_height="@dimen/next_lecture_select_width"
-            android:layout_margin="1dp"
-            android:background="@drawable/next_lecture_select"
-            android:text="4"
-            android:textColor="#ffffffff"
-            tools:ignore="HardcodedText" />
-
-    </LinearLayout>
+    <TextView
```
Comment (jacqueline8711) [inline app/src/main/res/layout/layout_lecture_selection.xml]: I don't see any usages for this layout, is it still needed?

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/studycard/StudyCardListAdapter.java [hunk 3]
```diff
@@ -58,7 +58,7 @@ public View getView(int position, View view, ViewGroup parent) {
         }
 
         if (card != null) {
-            // viewHolder.title.setText(card.getTitle());
+            // viewHolder.title.setText(cardView.getTitle());
```
Comment (jacqueline8711) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/studycard/StudyCardListAdapter.java]: This still says cardView

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/studycard/CardsDetailActivity.java [hunk 4]
```diff
@@ -58,14 +58,14 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
-        /* if (!card.is_valid()) {
+        /* if (!cardView.is_valid()) {
             return null;
         }
         try {
             ChatMember chatMember = Utils.getSetting(this, Const.CHAT_MEMBER, ChatMember.class);
             final TUMCabeVerification v = TUMCabeVerification.create(this, chatMember);
             final Context c = this;
-            return TUMCabeClient.getInstance(c).addStudyCard(card, v);
+            return TUMCabeClient.getInstance(c).addStudyCard(cardView, v);
```
Comment (jacqueline8711) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/studycard/CardsDetailActivity.java]: This is still supposed to be card

File: app/src/main/res/layout/layout_lecture_selection.xml (hunk 2 above)
Comment (thellmund) [reply app/src/main/res/layout/layout_lecture_selection.xml]: Is removed now.

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/studycard/CardsDetailActivity.java (hunk 4 above)
Comment (thellmund) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/studycard/CardsDetailActivity.java]: Anything related to study cards is currently not used in the app, so I did not bother with it. 
@kordianbruck can the study cards stuff be removed?

**Pull request merged**